internal/kafka/producer_pool.go

```diff
@@ -0,0 +1,71 @@
+package kafka
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+
+	"github.com/gol4ng/logger"
+	kafkaconfluent "gopkg.in/confluentinc/confluent-kafka-go.v1/kafka"
+)
+
+type ProducerPool interface {
+	Close()
+	Produce(ctx context.Context, messages chan *Message)
+}
+
+type producerPool struct {
+	logger        logger.LoggerInterface
+	client        Client
+	number        int
+	numberRunning int32
+	waitGroup     sync.WaitGroup
+}
+
+// NewProducerPool initializes a new kafka producer pool
+func NewProducerPool(logger logger.LoggerInterface, client Client, number int) *producerPool {
+	return &producerPool{
+		logger:        logger,
+		client:        client,
+		number:        number,
+		numberRunning: 0,
+		waitGroup:     sync.WaitGroup{},
+	}
+}
+
+// Close stops the producerPool and its goroutines
+func (w *producerPool) Close() {
```

Comment:
⚠️ if we want to  really stop the goroutine we must add a `stopChan` for each started worker and send stop when we want 
to stop worker goroutine

Reply:
I just tried to but it looks more complex to me (because I have to maintain an array of channels).
I think this is a good case to use `sync.WaitGroup`.